### PR TITLE
[ProjectConverter] Split out ConversionSettings from ConversionMetadata

### PR DESCRIPTION
**Pull request by FredKSchott**
*A part of #217: "Refactor ProjectConverter to Fix URL Handling"*
*Builds off of #220: "[ProjectConverter] Clean up url-converter logic/layout"*

## Summary:

This PR begins to refactor ProjectMetadata by breaking out the static conversion settings from `ConversionMetadata` into a new config object called `ConversionSettings`.

### CODE REVIEW

File: src/base-converter.ts [hunk 1]
```diff
@@ -12,109 +12,67 @@
...
-   * cases they are accessed without importing. The presumption is that access
-   * is guarded by a conditional and replcing with `undefined` will safely
-   * fail the guard.
-   */
-  readonly referenceExcludes?: Iterable<string>;
+// NOTE(fks) 11-06-2017: LEGACY!!!
+// Left over from ConversionMetadata, when the settings were mixed in to the
+// converter and its cache/results. Refactored out in the final PR, but left
+// in for now to keep this PR manageable.
+export interface ProjectConverterInterface {
```
Comment (FredKSchott) [inline src/base-converter.ts]: Once BaseConverter becomes ProjectConverter, this separate interface definition will no longer be needed. We keep it here to satisfy the old `ConversionMetadata` interface.

File: src/conversion-settings.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  /**
+   * Namespace names used to detect exports.
+   */
+  readonly namespaces: ReadonlySet<string>;
+  /**
+   * Files to exclude from conversion (ie lib/utils/boot.html).
+   */
+  readonly excludes: ReadonlySet<string>;
+  /**
+   * Files to exclusively include in conversion. Defaults to all files HTML
```
Comment (usergenic) [inline src/conversion-settings.ts]: Remove "exclusively" if these includes are additive to "all files HTML imported somewhere in the project".  It reads like this replaces whatever is imported somewhere.

File: src/conversion-settings.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+   */
+  readonly includes: ReadonlySet<string>;
+  /**
+   * Namespace references (ie, Polymer.DomModule) to "exclude" be replacing
+   * the entire reference with `undefined`.
+   */
+  readonly referenceExcludes: ReadonlySet<string>;
+  /**
+   * Namespace references (ie, document.currentScript.ownerDocument) to
+   * "rewrite" be replacing the entire reference with the given Node.
```
Comment (usergenic) [inline src/conversion-settings.ts]: `be replacing` -> `by replacing`

File: src/conversion-settings.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+/**
+ * This is the partial, user-provided configuration that a ConversionSettings
+ * object is generated from. User values are processed and expected defaults are
+ * added.
+ */
+export interface PartialConversionSettings {
+  /**
+   * Namespace names used to detect exports. Namespaces declared in the
+   * code with an @namespace declaration are automatically detected.
```
Comment (usergenic) [inline src/conversion-settings.ts]:
```
* code with a `@namespace` declaration are automatically detected.
```
(need to `escape` the `@namespace`)

**Pull request merged**